File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/ReadPropertyListValueRequest.java
package org.opcfoundation.webserver.service.message.digitaltwin;

import org.opcfoundation.webserver.types.digitaltwin.ObjectServiceContext;
import org.opcfoundation.webserver.types.common.UaChildId;

import java.util.Set;

public class ReadPropertyListValueRequest {
    private final String id;
    private final Set<String> propertyIds;
    private final Set<UaChildId> subPropertyIds;
    private final ObjectServiceContext context;

    public ReadPropertyListValueRequest(
            ObjectServiceContext context,
            Set<String> propertyIds,
            Set<UaChildId> subPropertyIds)
    {
        this.id = context.getObjectId().getId();
        this.propertyIds = propertyIds;
        this.subPropertyIds = subPropertyIds;
        this.context = context;
    }

    public String getId() {
        return id;
    }

    public Set<String> getPropertyIds() {
        return propertyIds;
    }

    public Set<UaChildId> getSubPropertyIds() {
        return subPropertyIds;
    }

    public ObjectServiceContext getContext() {
        return context;
    }
}

File: opcua-webservice-node/src/server/service/message/digitialtwin/ReadPropertyListValueRequest.ts
import { UaChildId } from "../../../types/common/UaChildId";
import { ObjectServiceContext } from "../../../types/digitaltwin/ObjectServiceContext";

export class ReadPropertyListValueRequest {
    private readonly _id: string;
    private readonly _propertyIds: Set<string>;
    private readonly _subPropertyIds: Set<UaChildId>;
    private readonly _context: ObjectServiceContext;

    constructor(
        context: ObjectServiceContext,
        propertyIds: Set<string>,
        subPropertyIds: Set<UaChildId>,
    ) {
        this._id = context.objectId.id;
        this._propertyIds = propertyIds;
        this._subPropertyIds = subPropertyIds;
        this._context = context;
    }

    get id(): string {
        return this._id;
    }

    get propertyIds(): Set<string> {
        return this._propertyIds;
    }

    get subPropertyIds(): Set<UaChildId> {
        return this._subPropertyIds;
    }

    get context(): ObjectServiceContext {
        return this._context;
    }
}